server/chat-plugins/poll.ts

```diff
@@ -71,42 +101,48 @@ export class Poll {
 		const userid = user.id;
 
 		if (!(userid in this.voters) || !(ip in this.voterIps)) {
-			this.voters[userid] = 0;
-			this.voterIps[ip] = 0;
+			this.voters[userid] = [];
+			this.voterIps[ip] = [];
 		}
 
 		this.updateTo(user);
 	}
 
-	generateVotes() {
+	generateVotes(user: User | null) {
+		const empty = this.multiPoll ? `<i class="fa fa-square-o" aria-hidden="true"></i>` : `<i class="fa fa-circle-o" aria-hidden="true"></i>`;
+		const chosen = this.multiPoll ? `<i class="fa fa-check-square-o" aria-hidden="true"></i>` : `<i class="fa fa-dot-circle-o" aria-hidden="true"></i>`;
+
+		const pendingVotes = (user && this.pendingVotes[user.id]) || [];
 		const iconText = this.isQuiz ? '<i class="fa fa-question"></i> Quiz' : '<i class="fa fa-bar-chart"></i> Poll';
 		let output = `<div class="infobox"><p style="margin: 2px 0 5px 0"><span style="border:1px solid #6A6;color:#484;border-radius:4px;padding:0 3px">${iconText}</span> <strong style="font-size:11pt">${this.getQuestionMarkup()}</strong></p>`;
 		for (const [num, option] of this.options) {
-			output += `<div style="margin-top: 5px"><button class="button" style="text-align: left" value="/poll vote ${num}" name="send" title="Vote for ${num}. ${Chat.escapeHTML(option.name)}">${num}. <strong>${this.getOptionMarkup(option)}</strong></button></div>`;
```

Comment:
We should still use the old style if it's _not_ a multi-poll.

Reply:
Do we want to keep one-click submits for single polls?

Reply:
Yeah, I prefer the "chat doesn't scroll when hovering over a button" option.

Reply:
Sure.